sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/meta/provider/pubsub/PubsubJsonTableProvider.java

```diff
@@ -54,125 +47,29 @@ public String getTableType() {
-     *
-     * <p>This attribute has to conform to the same requirements as in {@link
-     * PubsubIO.Read.Builder#withTimestampAttribute}.
-     *
-     * <p>Short version: it has to be either millis since epoch or string in RFC 3339 format.
-     *
-     * <p>If the attribute is specified then event timestamps will be extracted from the specified
-     * attribute. If it is not specified then message publish timestamp will be used.
-     */
-    @Nullable
-    abstract String getTimestampAttribute();
-
-    /**
-     * Optional topic path which will be used as a dead letter queue.
-     *
-     * <p>Messages that cannot be processed will be sent to this topic. If it is not specified then
-     * exception will be thrown for errors during processing causing the pipeline to crash.
-     */
-    @Nullable
-    abstract String getDeadLetterQueue();
 
-    /**
-     * Pubsub topic name.
-     *
-     * <p>Topic is the only way to specify the Pubsub source. Explicitly specifying the subscription
-     * is not supported at the moment. Subscriptions are automatically created (but not deleted).
-     */
-    abstract String getTopic();
+    Schema schema = tableDefinition.getSchema();
+    String location = tableDefinition.getLocation();
+    Schema dataSchema = tableDefinition.getSchema();
 
-    /**
-     * Table schema, describes Pubsub message schema.
-     *
-     * <p>If {@link #getUseFlatSchema()} is not set, schema must contain exactly fields
-     * 'event_timestamp', 'attributes, and 'payload'. Else, it must contain just 'event_timestamp'.
-     * See {@linkA PubsubMessageToRow} for details.
-     */
-    public abstract Schema getSchema();
+    PubsubSchemaCapableIOProvider ioProvider = new PubsubSchemaCapableIOProvider();
+    Schema configurationSchema = ioProvider.configurationSchema();
 
-    static Builder builder() {
-      return new AutoValue_PubsubJsonTableProvider_PubsubIOTableConfiguration.Builder();
-    }
-
-    @AutoValue.Builder
-    abstract static class Builder {
-      abstract Builder setUseFlatSchema(boolean useFlatSchema);
-
-      abstract Builder setSchema(Schema schema);
-
-      abstract Builder setTimestampAttribute(String timestampAttribute);
-
-      abstract Builder setDeadLetterQueue(String deadLetterQueue);
-
-      abstract Builder setTopic(String topic);
+    Row configurationRow =
+        Row.withSchema(configurationSchema)
+            .withFieldValue("timestampAttributeKey", timestampAttributeKey)
+            .withFieldValue("deadLetterQueue", deadLetterQueue)
+            .build();
 
-      abstract PubsubIOTableConfiguration build();
+    try {
+      SchemaIO pubsubSchemaIO = ioProvider.from(location, configurationRow, dataSchema);
+      return PubsubIOJsonTable.withConfiguration(pubsubSchemaIO, schema);
+    } catch (Exception InvalidConfigurationException) {
+      throw new InvalidTableException("Invalid configuration of table");
```

Comment:
This should just catch `InvalidConfigurationException e`, not any `Exception`.

Also you should include the information from the original exception when re-throwing. Either use `e.getMessage()` to insert the original message into the new one, or add the exception as a cause (`new InvalidTableException(msg, e)`), or both.

I think that should fix the unit test that's failing.